R/peak-list-functions.R

```diff
@@ -187,3 +197,76 @@ combinePeaks <- function(x, intensityFun = base::mean,
         stop("m/z values of combined spectrum are not ordered")
     res
 }
+
+#' @title Reduce mass peaks from input spectra to consensus peaks
+#'
+#' @description
+#'
+#' `consensusPeaks` reduces the peaks from all input spectra to a set of
+#' *consensus* peaks, i.e. peaks present in a certain proportion of input
+#' spectra (which can be defined with parameter `minProp`).
+#'
+#' For details on the mass peak grouping across samples see [combinePeaks()].
+#'
+#' @param minProp `numeric(1)` with the minimal required proportion of input
+#'     spectra (peak matrices) a mass peak has to be present to be included
+#'     in the consensus peak matrix.
+#'
+#' @inheritParams combinePeaks
+#'
+#' @return
+#'
+#' Peaks `matrix` with m/z and intensity values representing the consensus peaks
+#' from the provided peak matrices.
+#'
+#' @author Johannes Rainer
+#'
+#' @family peak matrix combining functions
+#'
+#' @export
+#'
+#' @examples
+#'
+#' p1 <- cbind(mz = c(12, 45, 64, 70), intensity = c(10, 20, 30, 40))
+#' p2 <- cbind(mz = c(17, 45.1, 63.9, 70.2), intensity = c(11, 21, 31, 41))
+#' p3 <- cbind(mz = c(12.1, 44.9, 63), intensity = c(12, 22, 32))
+#'
+#' ## No mass peaks identical thus consensus peaks are empty
+#' consensusPeaks(list(p1, p2, p3))
+#'
+#' ## Reducing the minProp to 0.2. The consensus spectrum will contain all
+#' ## peaks
+#' consensusPeaks(list(p1, p2, p3), minProp = 0.2)
+#'
+#' ## With a tolerance of 0.1 mass peaks can be matched across spectra
+#' consensusPeaks(list(p1, p2, p3), tolerance = 0.1)
+#'
+#' ## Report the minimal m/z and intensity
+#' consensusPeaks(list(p1, p2, p3), tolerance = 0.1, intensityFun = min,
+#'     mzFun = min)
+consensusPeaks <- function(x, minProp = 0.5, intensityFun = base::mean,
```

Comment:
The first lines are very similar to `combinePeaks`. Should we create an internal function for them? Here the `timeDomain` argument is missing. Is that intended?

All in all the functions `combinePeaks` and `consensusPeaks` are very similar. Can't we combine them into one?